addons/mail/static/src/core/web/messaging_menu.js

```diff
@@ -66,6 +72,79 @@ export class MessagingMenu extends Component {
         }
     }
 
+    navigate(direction) {
+        if (this.notificationItems.length === 0) {
+            return;
+        }
+        const activeOptionId = this.state.activeIndex !== null ? this.state.activeIndex : 0;
+        let targetId = undefined;
+        switch (direction) {
+            case "first":
+                targetId = 0;
+                break;
+            case "last":
+                targetId = this.notificationItems.length - 1;
+                break;
+            case "previous":
+                targetId = activeOptionId - 1;
+                if (targetId < 0) {
+                    this.navigate("last");
+                    return;
+                }
+                break;
+            case "next":
+                targetId = activeOptionId + 1;
+                if (targetId > this.notificationItems.length - 1) {
+                    this.navigate("first");
+                    return;
+                }
+                break;
+            default:
+                return;
+        }
+        this.state.activeIndex = targetId;
+        this.notificationItems[targetId].scrollIntoView({ block: "nearest" });
```

Comment:
I agree it's not elegant but it's a workaround caused by the notification items not coming from one single list (some even come from specific ifs).

I would be open to a solution without dom related methods but I couldn't find one.

Reply:
This works fine. I get why it's usually preferred not to use DOM methods, but in this case this is quite straight-forward. Without using DOM methods, this would make the code more complicated just for the sake of not using DOM methods, but in practice they work the same. Simplest working code matters more, so DOM manipulation is good enough ;)